content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,281 @@
+
+This technique is liked by authors familiar with testing frameworks that provide parameterized or generated test cases out of the box, like NUnit or JUnit. However, some caution is needed when this approach is used. Test suites organized like this can become large and can flood the test output panel with many entries, making it unreadable or causing performance problems in client browsers.
+
+### Chai
+
+JavaScript kata should use [Chai](https://www.chaijs.com/) as the assertion library.
+
+:::note
+For simplicity, through the rest of this article it's assumed that the `chai` package is imported and `assert` is extracted from it with the following statement:
+
+```javascript
+const chai = require('chai');
+const assert = chai.assert;
+```
+
+This is a convention used in many JavaScript kata, but it's not a requirement, and authors can choose to import the package in any way they find suitable for them. Also, the `should` or `expect` style can also be used from `chai`.
+:::
+
+The most commonly used assertions include:
+
+- [`assert.strictEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_strictequal) - Asserts that `actual === expected`.
+- [`assert.deepEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_deepequal) - Asserts that two arrays or objects have the same values. `strictEqual` does not work since arrays and objects are reference types in JS.
+- [`assert.closeTo(actual, expected, delta[, message])`](https://www.chaijs.com/api/assert/#method_closeto) - Asserts that `abs(actual - expected) <= delta`. Should be used when comparing floating-point numbers.
+
+See the [Chai docs](https://www.chaijs.com/api/assert/) for all the methods.
+
+:::note
+The [`truncateThreshold` option](https://www.chaijs.com/guide/styles/#configtruncatethreshold) controls how arrays are stringified when `expected` and `actual` values are prepared for display in assertion messages. Setting it to 0 provides the most detailed feedback even for large arrays, but then assertion messages can flood the console and make the output difficult to read.
+
+```javascript
+chai.config.truncateThreshold = 0;
+```
+
+:::
+
+### Generating random numbers
+
+JavaScript provides a simple [`Math.random`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) function that returns a random floating-point number between `0` and `1`. You can use this to generate random numbers by multiplying it by another number, like `5`, and then flooring it, which returns a random number between `0` and **4**. Here's a simple utility to generate a random number between a lower and upper limit, inclusive:
+
+```javascript
+function random(lower, upper) {
+  return Math.floor(Math.random() * (upper - lower + 1) + lower);
+}
+
+random(1, 10); // inclusive random number between 1 and 10
+```
+
+However, to make this simpler, the [lodash](https://lodash.com) library includes some helpful functions to ease this:
```

Comment:
```suggestion
However, to make this simpler, the [lodash](https://lodash.com) library includes some helpful functions:
```

Reply:
I find this sentence somewhat redundant, but please do not take my wording and grammar suggestions too seriously :)